janus_core/cli/phonons.py

```diff
@@ -0,0 +1,263 @@
+from janus_core.cli.types import (
+    Architecture,
+    CalcKwargs,
+    Device,
+    LogPath,
+    MinimizeKwargs,
+    ReadKwargs,
+    StructPath,
+    Summary,
+)
+from janus_core.cli.utils import (
+    check_config,
+    end_summary,
+    parse_typer_dicts,
+    start_summary,
+    yaml_converter_callback,
+)
+from janus_core.helpers.utils import dict_paths_to_strs
+
+app = Typer()
+
+
+@app.command(help="Calculate phonons and save results.")
+@use_config(yaml_converter_callback)
+def phonons(
+    # pylint: disable=too-many-arguments,too-many-locals
+    # numpydoc ignore=PR02
+    ctx: Context,
+    struct: StructPath,
+    struct_name: Annotated[
+        Optional[str],
+        Option(help="Name of structure name."),
+    ] = None,
+    supercell: Annotated[
+        list[int],
+        Option(help="Supercell lattice vectors."),
+    ] = None,
+    displacement: Annotated[
+        float,
+        Option(help="Displacement for force constants calculation, in A."),
+    ] = 0.01,
+    thermal: Annotated[
+        bool,
+        Option(help="Whether to calculate thermal properties."),
+    ] = False,
+    temp_start: Annotated[
+        float,
+        Option(help="Start temperature for CV calculations, in K."),
+    ] = 0.0,
+    temp_end: Annotated[
+        float,
+        Option(help="End temperature for CV calculations, in K."),
+    ] = 1000.0,
+    temp_step: Annotated[
+        float,
+        Option(help="Temperature step for CV calculations, in K."),
+    ] = 50,
+    dos: Annotated[
+        bool,
+        Option(help="Whether to calculate the DOS."),
+    ] = False,
+    pdos: Annotated[
+        bool,
+        Option(
+            help="Whether to calculate the PDOS.",
+        ),
+    ] = False,
+    minimize: Annotated[
+        bool, Option(help="Whether to minimize structure before calculations.")
+    ] = False,
+    fmax: Annotated[
+        float, Option(help="Maximum force for optimization convergence.")
+    ] = 0.1,
+    minimize_kwargs: MinimizeKwargs = None,
+    arch: Architecture = "mace_mp",
+    device: Device = "cpu",
+    read_kwargs: ReadKwargs = None,
+    calc_kwargs: CalcKwargs = None,
+    file_prefix: Annotated[
+        Optional[Path],
+        Option(
+            help=(
+                """
+                Prefix for output filenames. Default is inferred from structure name,
+                or chemical formula.
+                """
+            ),
+        ),
+    ] = None,
+    log: LogPath = "phonons.log",
+    summary: Summary = "phonons_summary.yml",
+):
+    """
+    Perform phonon calculations and write out results.
+
+    Parameters
+    ----------
+    ctx : Context
+        Typer (Click) Context. Automatically set.
+    struct : Path
+        Path of structure to simulate.
+    struct_name : Optional[PathLike]
+        Name of structure to simulate. Default is inferred from filepath or chemical
+        formula.
+    supercell : List[int]
```

Comment:
It isn't strictly a `list`, it can also be a single `int`.

Reply:
typer doesn't allow Union types yet, so even a single integer would be passed as [x]

Doesn't matter too much if we do change it to a string though.